modules/vm/main.tf

```diff
@@ -15,6 +15,7 @@ resource "aws_security_group_rule" "tfe_ui" {
   to_port                  = 443
   protocol                 = "tcp"
   source_security_group_id = var.aws_lb
+  cidr_blocks              = var.aws_lb == null ? ["0.0.0.0/0"] : null
```

Comment:
I think these CIDR blocks on these security group rules should be scoped to the subnet of the load balancer rather than wide open.